sources/packages/backend/apps/workers/src/services/disbursement-schedule/disbursement-schedule.service.ts

```diff
@@ -0,0 +1,179 @@
+import { Injectable } from "@nestjs/common";
+import {
+  ApplicationStatus,
+  DisbursementSchedule,
+  MSFAANumber,
+  OfferingIntensity,
+  RecordDataModelService,
+} from "@sims/sims-db";
+import { CustomNamedError } from "@sims/utilities";
+
+import { DataSource, IsNull, Not } from "typeorm";
+import { MSFAANumberService } from "..";
+import {
+  APPLICATION_MSFAA_ALREADY_ASSOCIATED,
+  DISBURSEMENT_NOT_FOUND,
+} from "../../constants";
+
+@Injectable()
+export class DisbursementScheduleService extends RecordDataModelService<DisbursementSchedule> {
+  constructor(
+    dataSource: DataSource,
+    private readonly msfaaNumberService: MSFAANumberService,
+  ) {
+    super(dataSource.getRepository(DisbursementSchedule));
+  }
+
+  /**
+   * Associates an MSFAA number to the disbursement(s) checking
+   * whatever is needed to create a new MSFAA or use an
+   * existing one instead.
+   * @param assessmentId assessment id of the workflow instance.
+   */
+  async associateMSFAANumber(assessmentId: number): Promise<void> {
+    const [firstDisbursement, secondDisbursement] = await this.repo.find({
+      select: {
+        id: true,
+        studentAssessment: {
+          id: true,
+          application: { id: true, student: { id: true } },
+          offering: { id: true, offeringIntensity: true, studyStartDate: true },
+        },
+        msfaaNumber: {
+          id: true,
+        },
+      },
+      relations: {
+        studentAssessment: { application: { student: true }, offering: true },
+        msfaaNumber: true,
+      },
+      where: {
+        studentAssessment: { id: assessmentId },
+      },
+    });
+
+    if (!firstDisbursement) {
+      throw new CustomNamedError(
+        "Disbursement not found or one of its associations is missing.",
+        DISBURSEMENT_NOT_FOUND,
+      );
+    }
+
+    if (firstDisbursement.msfaaNumber?.id) {
+      throw new CustomNamedError(
```

Comment:
I have 2 commnets, 
1. In this PR, we are not considering the case of `reactivating` the msfaa by the federal right? 
2. I assume this is the method that is called by the camunda to associate the msfaa number for the first time, so, I would like to know what is the possible scenario, where msfaa number will already be associated with the disbursement.

Reply:
More information about workers best practices: https://docs.camunda.io/docs/components/best-practices/development/writing-good-workers/

Reply:
@ann-aot 

1. Yes we are not considering reactivating as just groomed yesterday and importantly is part of #1777 
2. As @andrewsignori-aot mentioned it is to ensure the idempotency. 

I was also thinking about checking socendDisbursement.msfaaNumber?.id but realized that the association either happens to both or none. So didn't consider it.